### ISSUE TITLE
Updating the "PredPreyRatioMax" field of FishBase/SeaLifeBase's ESTIMATE table

### ISSUE THREAD

**Issue by agruss2** [jhpoelen/fb-osmose-bridge#151]
@Dengaloo @FIN-JBarile @FIN-casey
The present issue is related to #150 

As described in #32 and in this document ([OSMOSE_Methods_Predator_prey_size_ratios.docx](https://github.com/jhpoelen/fb-osmose-bridge/files/1565278/OSMOSE_Methods_Predator_prey_size_ratios.docx)), I developed generalized additive models (GAMs) that predict maximum predator/prey size ratios.
We now need to update the "PredPreyRatioMax" field of FishBase/SeaLifeBase's ESTIMATE table, using the predictions of the GAMs that I developed.
We should proceed as follows:
**(1)** If the species under consideration belongs to one of the following orders, then the "complex" GAM of maximum predator/prey size ratio should be used to estimate the maximum predator/prey size ratio of the species: Carcharhiniformes, Chimaeriformes, Clupeiformes, Decapoda, Elopiformes, Euphausiacea, Gadiformes, Mugiliformes, Myctophiformes, Myliobatiformes, Octopoda, Osmeriformes, Perciformes, Pleuronectiformes, Salmoniformes, Scorpaeniformes, Sepiida, Siluriformes, Squaliformes, Stomatopoda, Stomiiformes, and Teuthida.
**(2)** If the species under consideration does not belong to an order listed in **(1)** BUT has a trophic level (TL) provided in FishBase or SeaLifeBase, then the "simple" GAM of maximum predator/prey size ratio should be used to estimate the maximum predator/prey size ratio of the species.
**(3)** Finally, if the species under consideration does not belong to an order listed in (1) AND does not have a trophic level (TL) provided in FishBase or SeaLifeBase, then the maximum predator/prey size ratio of the species should be set to 30.0.

To figure out how to update the "PredPreyRatioMax" field of FishBase/SeaLifeBase's ESTIMATE table from the predictions of the GAMs that I developed, I suggest that we meet on Skype soon. I will be available any time from now to December, 22nd and then again starting from December, 31st. Please let me know a day and time where we could meet.
Many thanks!

**Comment by agruss2** [comment]
@Dengaloo @FIN-JBarile As discussed during our last Skype meeting, I am now in charge of updating FB's and SLB's ESTIMATE tables. 
However, making predictions with the generalized additive models (GAMs) that I developed is taking me far much time than initially anticipated. 
I should not be done with this task before some time next week. Once I am done, I will send you the updated FB's and SLB's ESTIMATE tables in the form of spreadsheets, which I will ask you to integrate in FB and SLB. Finally, I will be able to run some tests to see if the updated PredPreyRatioMax fields of the FB's and SLB's ESTIMATE tables are satisfactory or if some ultimate modifications are needed. 
Many thanks!

**Comment by agruss2** [comment]
@FIN-JBarile @QQ-Sortiz (CC: @Dengaloo)
I am done updating the PredPreyRatioMin and PredPreyRatioMax fields of the ESTIMATE tables, based on the predictions of my generalized additive models. 
Here is the updated FB's ESTIMATE table:  [ESTIMATE_v3.xlsx](https://github.com/jhpoelen/fb-osmose-bridge/files/1671786/ESTIMATE_v3.xlsx)
And here is the updated SLB's ESTIMATE table: [ESTIMATE_SLB_v3.xlsx](https://github.com/jhpoelen/fb-osmose-bridge/files/1671787/ESTIMATE_SLB_v3.xlsx)
Could you know please update your ESTIMATE tables (those that the API use to derive parameter estimates for OSMOSE)? 
Once you are done with these updates, please post a comment here to let me know. I will then run a test for some ecosystems to find out whether the predator/prey size ratio estimates provided by the API are satisfactory. 
Many thanks!

**Comment by agruss2** [comment]
@QQ-Sortiz 
(CC: @Dengaloo @FIN-JBarile @jhpoelen) 
This is has become an API issue, and you are the responsible for API issues. 
Therefore, could you please solve this issue by March, 11th? If needed, please ask questions to @jhpoelen 
Thanks a lot.

**Comment by QQ-Sortiz** [comment]
Hello @agruss2,

I tried to change estimates table for FishBase/SeaLifeBase on the API. Please check.

**Comment by agruss2** [comment]
@QQ-Sortiz (CC: @FIN-JBarile)  
Many thanks for having worked on this!
I ran a test where I queried OSMOSE parameters for six different ecosystems. 
In all cases, I obtained a maximum predator/prey size ratio of 30 for all focal functional groups. In other words, in all cases, the maximum predator/prey size ratio of all focal functional groups was set to the default value of the parameter. 
This should not be the case, because of the large number and diversity of maximum predator/prey size ratios in the "ESTIMATE_v3.xlsx" and "ESTIMATE_SLB_v3.xlsx" files that I shared above. 
I suspect that what is happening is that the API is currently setting all maximum predator/prey size ratios to 30 instead of extracting relevant values from FishBase's and SeaLifeBase's ESTIMATE tables. 
Could you please correct for this?
Thank you very much.

**Comment by jhpoelen** [comment]
@QQ-Sortiz looks great - you might want to consider adding the "table_names.tsv" to https://github.com/jhpoelen/fishbase_archiver/releases/tag/v0.2.1-patch3 . Also, I'd suggest to make reference in the release description to the issue in which the reason for the patch was discussed.

**Comment by agruss2** [comment]
@QQ-Sortiz 
I ran a test where I queried parameters for OSMOSE models for different ecosystems. The maximum predator/prey size ratio estimates I obtained were as expected and made sense. 
Many thanks for your very good work on this!!
I am now going to close the present issue.
The next step will be for me to: (1) query OSMOSE parameters for all ecosystems, in order to find out whether an "osmose_config.zip" file can be obtained for all ecosystems (and not only JSON files for the ecosystems characterized by a very high biodiversity); and (2) examine in great detail the parameter values provided for several chosen ecosystems, so as to determine whether some things need to be tweaked in the bridge between FB/SLB and OSMOSE. 
Meanwhile, I will start working on the first draft of manuscript.
Talk very soon!
